packages/dds/shared-object-base/src/types.ts

```diff
@@ -53,7 +53,19 @@ export interface ISharedObjectEvents extends IErrorEvent {
 }
 
 /**
- * Base interface for shared objects from which other interfaces derive. Implemented by SharedObject
+ * Base interface for shared objects from which other interfaces extend.
+ * @remarks
+ * This interface is not intended to be implemented outside this repository:
+ * implementers should migrate to using an existing implementation instead.
+ * @privateRemarks
+ * Implemented by {@link SharedObjectCore}.
+ *
+ * TODO:
+ * The relationship between the "shared object" abstraction and "channel" abstraction should be clarified and/or unified.
+ * Either there should be a single named abstraction or the docs here need to make it clear why adding events and bindToContext to a channel makes it a "shared object".
+ * Additionally the docs here need to define what a shared object is, not just claim this interface is for them.
+ * If the intention is that the "shared object" concept `IFluidLoadable` mentions is only ever implemented by this interface then even more concept unification should be done.
+ * If not then more clarity is needed on what this interface specifically is, what the other "shared object" concept means and how they relate.
```

Comment:
These are all great avenues to explore.  I think this is a mix of bad names that were never revisited and dreams of more-general applicability that were never realized.